viadot/sources/velux_club.py

```diff
@@ -0,0 +1,186 @@
+    ) -> str:
+        if source in ["jobs", "product", "company"]:
+            # check if date filter was passed!
+            if from_date == "" or to_date == "":
+                return "Introduce a 'FROM Date' and 'To Date'"
+            url = f"{api_url}{source}?from={from_date}&to={to_date}&region&limit=100"
+        elif source in "survey":
+            url = f"{api_url}{source}?language=en&type=question"
+        else:
+            return "pick one these sources: jobs, product, company, survey"
+        return url
+
+    def get_response(
+        self,
+        source: str = "",
+        from_date: str = "2022-03-22",
+        to_date: str = "",
+        region="null",
+    ) -> pd.DataFrame:  ## Returns the response
+        # DocString
+
+        # Dealing with bad arguments
+        if (source not in ["jobs", "product", "company", "survey"]):
+            raise Source_NOK("The source has to be: jobs, product, company or survey")
+
+        from_date_obj = datetime.strptime(from_date, '%Y-%m-%d')  
+        oldest_date_obj = datetime.strptime("2022-03-22", '%Y-%m-%d')           
+        delta = from_date_obj - oldest_date_obj  
+        
+        if (delta.days <0 ):
+            raise Historical_Too_Old("from_date has to be earlier than 2023-03-22!!")
+
+        to_date_obj = datetime.strptime(to_date, '%Y-%m-%d')  
+        delta = to_date_obj - from_date_obj  
+
+        if (delta.days <0 ):
+            raise Dates_NOK("to_date has to be earlier than from_date!")
+        
+
+        # Prepering the Query
+        url = self.build_query(source, from_date, to_date, self.API_URL, region)
+        headers = self.headers
+
+        # Getting first page
+        response = requests.request("GET", url, headers=headers)
+
+        # Next Pages
+        response = response.json()
+
+        if isinstance(response, dict):
+            keys_list = list(response.keys())
+        elif isinstance(response, list):
+            keys_list = list(response[0].keys())
+        else:
+            keys_list = []
+
+        if "data" in keys_list:
+            # first page content
+            df = pd.DataFrame(response["data"])
+            # next pages
+            while response["next_page_url"] != None:
+                url = f"{response['next_page_url']}&from={from_date}&to={to_date}&region&limit=100"
+                r = requests.request("GET", url, headers=self.headers)
+                response = r.json()
+                df_page = pd.DataFrame(response["data"])
+                if source == "product":
+                    df_page = df_page.T
+
+                df = pd.concat((df, df_page), axis=0)
+        else:
+            df = pd.DataFrame(response)
+
+        return df
+
+    def to_parquet(
+        self,
+        response,
+        path="",
+        if_exists: Literal["append", "replace", "skip"] = "replace",
+        **kwargs,
+    ) -> None:
+        # DocString
+        if path == "":
+            raise SourceNOK_Error
```

Comment:
1) Don't use empty string as a default  value
2) The standard way is to throw a `ValueError` when user provided an incorrect param